File: Go/ExampleTests/common/patchtitle_string.go
package common

type PatchTitleString struct {
	Title string
}

func NewPatchTitleStringFromSlice(v []string) PatchTitleString {
	s := PatchTitleString{}
	if len(v) > 0 { s.Title = v[0] }
	return s
}

// NewPatchTitleStringFromText builds from the text form, e.g. Money as "25 USD".
func NewPatchTitleStringFromText(text string) PatchTitleString {
	parts := RequireTokens(text, 1, "PatchTitle")
	return PatchTitleString{
		Title: parts[0],
	}
}

func (s PatchTitleString) String() string {
	return Token(s.Title)
}

func (s PatchTitleString) Equals(o PatchTitleString) bool {
	return DNCEqual(s.Title, o.Title)
}

func EqualPatchTitleStringSlices(a, b []PatchTitleString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/patchtitle_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct PatchTitleString {
    pub title: String,
}

impl PatchTitleString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            title: v.get(0).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 1, "PatchTitle");
        Self {
            title: parts[0].clone(),
        }
    }
}

impl std::fmt::Display for PatchTitleString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.title)].join(" "))
    }
}

impl PartialEq for PatchTitleString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.title, &other.title)
    }
}
